impeller/geometry/vector.h

```diff
@@ -14,27 +15,46 @@
 
 namespace impeller {
 
-struct Vector3 {
+template <class T, class U>
```

Comment:
As it exists this is a really dangerous implementation because TVector<Half> a = TVector<Scalar>.SomeMethod() will truncate to zero. We need to replace all of the static cast calls everywhere with Cast